lib/node/message.ml

```diff
@@ -5,16 +5,30 @@ type category =
   | Uncategorized
   | Request
   | Response
+  | Post
```

Comment:
Following Deku integration, this category is totally internal.
Since we only expose `create_request`, `create_response` and `create_post` some categories are never going to be used (`Uncategorized` and `Custom`).

How about getting rid of these useless types, and rename the `sub_category_opt` in the other branch to let the client set it?

Reply:
It's used internally to know how to process messages though? What exactly is your suggestion?

Reply:
Sorry to only say this after you've gone on vacation, but I'm not sure I really follow this design/interface; hopefully we can have a good discussion about it when you're back, or if Maria has the same idea I can talk to her about it :smile:.